### PR TITLE
FUSETOOLS-2144 - ensuring that the list of supported models is

### PR DESCRIPTION
**Pull request by bfitzpat**
initialized before being iterated through.
- added code to ensure initialization
- added unit test to test the method, which wasn't being tested
previously

### CODE REVIEW

File: core/tests/org.fusesource.ide.camel.model.service.core.tests/src/test/java/org/fusesource/ide/camel/model/service/core/catalog/CamelModelFactoryTest.java [hunk 1]
```diff
@@ -27,6 +27,12 @@
 																		 "2.15.1",
 																		 "2.17.0.redhat-630077", 
 																		 "2.17.0.redhat-630159");
+	@Test
+	public void testGetLatestCamelVersion() {
```
Comment (apupier) [inline core/tests/org.fusesource.ide.camel.model.service.core.tests/src/test/java/org/fusesource/ide/camel/model/service/core/catalog/CamelModelFactoryTest.java]: camel model factory are initialized by extension points, so this test needs to be an integration one.

File: core/tests/org.fusesource.ide.camel.model.service.core.tests/src/test/java/org/fusesource/ide/camel/model/service/core/catalog/CamelModelFactoryTest.java [hunk 2]
```diff
@@ -27,6 +27,12 @@
...
+	    String version = CamelModelFactory.getLatestCamelVersion();
+	    assertThat(version).isNotNull();
```
Comment (apupier) [inline core/tests/org.fusesource.ide.camel.model.service.core.tests/src/test/java/org/fusesource/ide/camel/model/service/core/catalog/CamelModelFactoryTest.java]: no need to assert not null as we assert just after that it is equals to 2.17.3

File: core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/catalog/CamelModelFactory.java [hunk 3]
```diff
@@ -181,6 +181,10 @@ public static String getCompatibleCamelVersion(String requestedCamelVersion) {
 	 */
 	public static String getLatestCamelVersion() {
 		String latest = null;
+		// make sure we're initialized here
+		if (supportedCamelModels == null || supportedCamelModels.keySet() == null) {
+		    initializeModels();
```
Comment (apupier) [inline core/plugins/org.fusesource.ide.camel.model.service.core/src/org/fusesource/ide/camel/model/service/core/catalog/CamelModelFactory.java]: if called from several places, I think that we need to create a singleton and have a synchronized block

**Comment by bfitzpat** [comment]
I'm going to close this. Turns out it was an issue with the test on the SY side (not surprising) and not on the Fuse side.

**Pull request closed without merging**